### PR TITLE
Task/7 data for health EndPoint

### PR DESCRIPTION
**Pull request by ThomasObenaus**
With this PR the cosmos connector provides the data needed for doing health checks.

### CODE REVIEW

**Comment by ThomasObenaus** [comment]
This PR closes #7

File: pool_test.go [hunk 1]
```diff
@@ -13,6 +15,105 @@ import (
...
+			time.Sleep(time.Millisecond * time.Duration(millies))
+		}()
+	}
+	wg.Wait()
+	ticker.Stop()
+	pool.Close()
+}
+
+func TestIsConnected(t *testing.T) {
+	// GIVEN
```
Comment (frank-e-zimmer) [inline pool_test.go]: In my opinion too many tests in one test method.

Comment (ThomasObenaus) [reply pool_test.go]: I did a split

File: pool_test.go [hunk 2]
```diff
@@ -13,6 +15,105 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestIsConnectedRace(t *testing.T) {
+	// This test shall detect data races when
```
[the remark this answers is not in the document]
Comment (ThomasObenaus) [reply pool_test.go]: Added docu

File: pool.go [hunk 3]
```diff
@@ -77,9 +77,35 @@ type idleConnection struct {
 	idleSince time.Time
 }
 
+// IsConnected return true in case at least one (idle or active) connection
```
[the remark this answers is not in the document]
Comment (ThomasObenaus) [reply pool.go]: fixed

File: examples/local/main.go [hunk 4]
```diff
@@ -25,32 +25,48 @@ func main() {
...
-			}
-		}
-	}()
+	exitChannel := make(chan struct{})
+	go processLoop(cosmos, logger, exitChannel)
 
-	<-exit_chan
+	<-exitChannel
 	if err := cosmos.Stop(); err != nil {
 		logger.Error().Err(err).Msg("Failed to stop")
```
[the remark this answers is not in the document]
Comment (ThomasObenaus) [reply examples/local/main.go]: Fixed

File: cosmos_test.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+	cosmos, err := New("ws://host")
+	require.NoError(t, err)
+	require.NotNil(t, cosmos)
+	cosmos.pool = mockedQueryExecutor
+
+	// WHEN
+	mockedQueryExecutor.EXPECT().Close().Return(nil)
+	err = cosmos.Stop()
+	assert.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (ThomasObenaus) [reply cosmos_test.go]: I added it

File: cosmos_test.go [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockedQueryExecutor := mock_interfaces.NewMockQueryExecutor(mockCtrl)
+
+	cosmos, err := New("ws://host")
+	require.NoError(t, err)
+	require.NotNil(t, cosmos)
+	cosmos.pool = mockedQueryExecutor
+
+	// WHEN -- connected --> healthy
```
[the remark this answers is not in the document]
Comment (ThomasObenaus) [reply cosmos_test.go]: Fixed

**Pull request merged**